### DISCUSSION TITLE
Question with query_key

### DISCUSSION THREAD

**Discussion by lihuanan** [jertel/elastalert2#788] [Q&A] [answered]
Hi member

Using elastalert2，I met a question in using query_key.
for example:
I have multiple 'service' and 'group' 
When any service received more than 1 "400" or "500" errors, I want an alert, then I have:

name:  the statuscode error
type: frequency
index: testindex
num_events: 1
timeframe:
  minutes: 5
query_key: service,group
filter:
- range:
    response:
        from: 400
		
alert:
- "email"
email:
- "myemail@email.com"

alert_text_type: alert_text_only

alert_text: |
    service: {}
    group: {}
    time range: {:%Y-%m-%d %H:%M:%S} to {:%Y-%m-%d %H:%M:%S}
    query: {}
    num_events: {}
    event_occured: {}

alert_text_args:
    - service
    - group
    - original_starttime
    - endtime
    - filter
    - num_events
    - num_hits/num_match

I use query_key to get alert independently for 'apple,g1' OR 'orange,g2' OR 'apple,g2'
I want to get event_occured for  'apple,g1' OR 'orange,g2' OR 'apple,g2' alone, too.
Unfortunately I can only find 2 useful number args in _alerts.rule or _alerts.match: num_matches and num_hits.
num_hits is the total hits of all querys, including all the query_key
#Q1: how can I get the hits for 'apple,g1' OR 'orange,g2' OR 'apple,g2' alone?
#Q2: it will send 3 alarms, how to send 1 alarm together

**Comment by jertel** [comment, marked as answer]
Please read #11 before requesting support.

> #Q1: how can I get the hits for 'apple,g1' OR 'orange,g2' OR 'apple,g2' alone?

This isn't supported natively. To write your own enhancement to calculate values, see https://github.com/jertel/elastalert2/discussions/717 for a similar question.

> #Q2: it will send 3 alarms, how to send 1 alarm together

The [aggregation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#aggregation) configuration might be useful.
